File: lab7/Plant.java

public class Plant {

	protected String latin_name;
	protected int lifespan;
	private static int count = 0;
	
	public String get_latin_name()
	{
		return latin_name;
	}
	public void set_latin_name(String latin_name)
	{
		this.latin_name = latin_name;
	}
	
	public int get_lifespan()
	{
		return lifespan;
	}
	public void set_lifespan(int lifespan)
	{
		this.lifespan = lifespan;
	}
	
	public static int get_count()
	{
		return count;
	}
	
	public Plant()
	{
		latin_name = "Rosa majalis";
		lifespan = 5;
		count++;
	}
	public Plant(Plant plant)
	{
		latin_name = plant.latin_name;
		lifespan = plant.lifespan;
		count++;
	}
	protected void finalize()
	{
		count--;
	}
}

File: lab7 - c++ java/Plant.cpp
#include "Plant.h"

int Plant::count = 0;

std::string Plant::get_latin_name()
{
	return latin_name;
}
void Plant::set_latin_name(std::string latin_name)
{
	this->latin_name = latin_name;
}

int Plant::get_lifespan()
{
	return lifespan;
}
void Plant::set_lifespan(int lifespan)
{
	this->lifespan = lifespan;
}

Plant::Plant()
{
	latin_name = "Rosa villosa";
	lifespan = 5;
	count++;
}

Plant::Plant(const Plant& plant)
{
	latin_name = plant.latin_name;
	lifespan = plant.lifespan;
	count++;
}
Plant::~Plant()
{
	count--;
}